BaseClasses.py

```diff
@@ -776,9 +776,11 @@ def _update_reachable_regions_explicit_indirect_conditions(self, player: int, qu
                 self.path[new_region] = (new_region.name, self.path.get(connection, None))
 
                 # Retry connections if the new region can unblock them
-                for new_entrance in self.multiworld.indirect_connections.get(new_region, set()):
-                    if new_entrance in blocked_connections and new_entrance not in queue:
-                        queue.append(new_entrance)
+                entrances = self.multiworld.indirect_connections.get(new_region)
+                if entrances is not None:
+                    entrances = entrances.intersection(blocked_connections)
```

Comment:
```suggestion
                    entrances.intersection_update(blocked_connections)
```
nitpicky, admittedly, but if we're using  _update one line below, I feel this looks better.

Reply:
A new set needs to be created here because `entrances` starts off as a set instance that is stored within `self.multiworld.indirect_connections`, which must not be modified by this function.

I could make a new variable for better clarity.
`relevant_entrances = entrances.intersection(blocked_connections)`

Reply:
Would it be better to copy it once at the start then, so that you can modify the instance?

Reply:
No. I don't see how copying the set would help.

`first_set.intersection(second_set)` only needs to create a new empty set, iterate the smaller of the two sets, and add the iterated elements that are in the larger set into the new empty set.

Copying the set should take more time and memory.

Reply:
Oh, I thought this worked differently, nvm